src/Orleans.Runtime/GrainDirectory/MultiClusterRegistration/GlobalSingleInstanceActivationMaintainer.cs

```diff
@@ -16,33 +17,28 @@ namespace Orleans.Runtime.GrainDirectory
     internal class GlobalSingleInstanceActivationMaintainer : SingleTaskAsynchAgent
     {
         private readonly object lockable = new object();
-        private readonly GlobalConfiguration config;
         private readonly LocalGrainDirectory router;
         private readonly Logger logger;
         private readonly IInternalGrainFactory grainFactory;
         private readonly TimeSpan period;
         private readonly IMultiClusterOracle multiClusterOracle;
+        private readonly SiloOptions siloOptions;
+        private readonly MultiClusterOptions multiClusterOptions;
 
         // scanning the entire directory for doubtful activations is too slow.
         // therefore, we maintain a list of potentially doubtful activations on the side.
         // maintainer periodically takes and processes this list.
         private List<GrainId> doubtfulGrains = new List<GrainId>();
 
-        public GlobalSingleInstanceActivationMaintainer(
-            LocalGrainDirectory router,
-            Logger logger,
-            GlobalConfiguration config,
-            IInternalGrainFactory grainFactory,
-            IMultiClusterOracle multiClusterOracle,
-            ExecutorService executorService,
-            ILoggerFactory loggerFactory)
+        public GlobalSingleInstanceActivationMaintainer(LocalGrainDirectory router, Logger logger, GlobalConfiguration config, IInternalGrainFactory grainFactory, IMultiClusterOracle multiClusterOracle, ExecutorService executorService, IOptions<SiloOptions> siloOptions, IOptions<MultiClusterOptions> multiClusterOptions, ILoggerFactory loggerFactory)
```

Comment:
the multi-line format makes merging & diffs simpler

Reply:
I tend to not collapse them and keep as they were. Probably some voodoo magic from ReSharper... will get back to multi-line